accepted/000070-cluster_awareness.md

```diff
@@ -0,0 +1,309 @@
+
+* CaaSP: /etc/products.d/caasp.prod + pattern `pattern-caasp-Node`
+* SES: /etc/products.d/ses.prod
+* SLE-HA/SHAP: /etc/products.d/SLES_SAP.prod, /etc/products.d/sle-ha.prod
+
+When a system is locked, no action will be executed. System lock may be disabled by the user by disabling the corresponding Salt Formula.
+
+It will be possible to define an allowed list of actions that Uyuni/SUSE Manager can issue on a specific node by implementing a combination of selective blackout (https://github.com/uyuni-project/uyuni-rfc/pull/31/) and package locking.
+
+A product can define a list of:
+- an allowed list of actions in Uyuni/SUSE Manager action terms
+- packages or patterns to lock at the minion level
+
+This information must be provided by the product and must be inserted into Uyuni/SUSE Manager database.
+Upon bootstrapping, Uyuni/SUSE Manager will issue the required package locks and the allowed list of actions will be not part of the blackout.
+Example:
+- CaaSP define the `zypper lock` the `patterns-caasp*` pattern
+- Uyuni/SUSE Manager allows package install, modify, removal actions
+
+#### Cluster node as special citizens in the minion domain
+
+All systems add-on types will be disabled for installation, except for the Monitoring add-on.
+When deleting a system, a cleanup of salt-minion must not be done (requested by SES).
+
+All other actions will be inhibited by the system lock and will require special handling by being implemented in the cluster provider manager level.
+
+#### System groups
+
+When defining a new cluster, Uyuni/SUSE Manager also creates a system group named `<type><name>` in which all nodes of the cluster will be inserted.
+
+## Integration requirements from an external product perspective
+
+From the plugin perspective, a product needs to define:
+- a new cluster type in Uyuni/SUSE Manager with a specific configuration (cluster type <-> cluster provider manager)
+- provide a system add-on type (or a Salt Formula) that provisions the management node with cluster management software
+
+When a new cluster product type is available in Uyuni/SUSE Manager, the association between the cluster type and its cluster provider manager is stored in the database.
+
+Summary of what a product must provide to Uyuni/SUSE Manager:
+
+* CaaSP:
+  - Salt Formula to install CaaSP software on the management node
+  - Salt Formula to install CaaSP software for control-plane/workers
+  - Cluster provider manager in a Salt module: skuba
+  - (Optional) Salt Formula to provision a load balancer
+* SES:
+  - Salt Formula to provision an SES minion
+  - Salt Formula to provision the management node
+  - Cluster provider manager in a Salt module: cephadm
+* SLE-HA/SHAP:
+  - Salt Formula to provision the management node
+  - Salt Formula to provision a SLE-HA/SHAP node
+  - Cluster provider manager in a Salt module: crm
+
+## Opening Uyuni/SUSE Manager features to cluster objects
```

Comment:
Nitpick: if his is not in the scope of this RFC I would suggest to move the paragraph in a "Next Steps" section.

Reply:
Done